From patchwork Wed Mar 31 15:06:56 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Masahiro Yamada <masahiroy@kernel.org>
X-Patchwork-Id: 12175581
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-17.0 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,
	SPF_HELO_NONE,SPF_PASS,USER_AGENT_GIT autolearn=unavailable
	autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id B87DEC433ED
	for <kvm@archiver.kernel.org>; Wed, 31 Mar 2021 15:18:32 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 7D25661025
	for <kvm@archiver.kernel.org>; Wed, 31 Mar 2021 15:18:32 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S236270AbhCaPSB (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Wed, 31 Mar 2021 11:18:01 -0400
Received: from condef-08.nifty.com ([202.248.20.73]:45436 "EHLO
        condef-08.nifty.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S236342AbhCaPRj (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 31 Mar 2021 11:17:39 -0400
X-Greylist: delayed 398 seconds by postgrey-1.27 at vger.kernel.org;
 Wed, 31 Mar 2021 11:17:38 EDT
Received: from conuserg-08.nifty.com ([10.126.8.71])by condef-08.nifty.com
 with ESMTP id 12VF7Uew010895
        for <kvm@vger.kernel.org>; Thu, 1 Apr 2021 00:07:51 +0900
Received: from localhost.localdomain (133-32-232-101.west.xps.vectant.ne.jp
 [133.32.232.101]) (authenticated)
        by conuserg-08.nifty.com with ESMTP id 12VF77DF021201;
        Thu, 1 Apr 2021 00:07:08 +0900
DKIM-Filter: OpenDKIM Filter v2.10.3 conuserg-08.nifty.com 12VF77DF021201
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=nifty.com;
        s=dec2015msa; t=1617203228;
        bh=ARbRFUFLNuhizfOq+rbCISdwf0ZvC9EkFJQtTKtjlps=;
        h=From:To:Cc:Subject:Date:From;
        b=0BXLesIP5v+3wtn3FQqyG2ppqrZdnq9H6gl0+IA5olp+NBiyCMNfCxmwDLQuR2eEI
         fl2kirvxTPzE5h+DkZRpO+kgavsmYNHxX6EA9+PAyIdGzYus1o3X6twC+Trr1bC9sJ
         XLYKukwdDHj7ON3/wu+41FoyMeD6G34UM93Cwa18iw8q+lICyienKgDyKE0iANeqDk
         bYfB0qMC99jkoUkhkEVUxfJIepDIsm7lV3ru0xALDquISA9GngQcrMXyd8ON/31gSm
         RzLmoIJDGR3HGPhB9AUota4bXQ/Tuswa/CExOvGPjpK9qDD5SNrH5OA6LSyNzF/uMV
         GY0ArNoN6PO3Q==
X-Nifty-SrcIP: [133.32.232.101]
From: Masahiro Yamada <masahiroy@kernel.org>
To: linux-mips@linux-mips.org
Cc: Masahiro Yamada <masahiroy@kernel.org>,
        Aleksandar Markovic <aleksandar.qemu.devel@gmail.com>,
        Huacai Chen <chenhuacai@kernel.org>,
        Thomas Bogendoerfer <tsbogend@alpha.franken.de>,
        kvm@vger.kernel.org, linux-kernel@vger.kernel.org,
        linux-mips@vger.kernel.org
Subject: [PATCH 1/2] mips: replace deprecated EXTRA_CFLAGS with ccflags-y
Date: Thu,  1 Apr 2021 00:06:56 +0900
Message-Id: <20210331150658.38919-1-masahiroy@kernel.org>
X-Mailer: git-send-email 2.27.0
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

As Documentation/kbuild/makefiles.rst says, EXTRA_CFLAGS is deprecated.
Replace it with ccflags-y.

Signed-off-by: Masahiro Yamada <masahiroy@kernel.org>
---

 arch/mips/kvm/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/mips/kvm/Makefile b/arch/mips/kvm/Makefile
index 506c4ac0ba1c..7d42d624a7b9 100644
--- a/arch/mips/kvm/Makefile
+++ b/arch/mips/kvm/Makefile
@@ -4,7 +4,7 @@
 
 common-objs-y = $(addprefix ../../../virt/kvm/, kvm_main.o coalesced_mmio.o eventfd.o)
 
-EXTRA_CFLAGS += -Ivirt/kvm -Iarch/mips/kvm
+ccflags-y += -Ivirt/kvm -Iarch/mips/kvm
 
 common-objs-$(CONFIG_CPU_HAS_MSA) += msa.o
 

From patchwork Wed Mar 31 15:06:57 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Masahiro Yamada <masahiroy@kernel.org>
X-Patchwork-Id: 12175583
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-17.0 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,
	SPF_HELO_NONE,SPF_PASS,USER_AGENT_GIT autolearn=unavailable
	autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 962E5C433B4
	for <kvm@archiver.kernel.org>; Wed, 31 Mar 2021 15:19:04 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 5278C60725
	for <kvm@archiver.kernel.org>; Wed, 31 Mar 2021 15:19:04 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S236091AbhCaPSg (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Wed, 31 Mar 2021 11:18:36 -0400
Received: from condef-07.nifty.com ([202.248.20.72]:24700 "EHLO
        condef-07.nifty.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S236026AbhCaPSZ (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 31 Mar 2021 11:18:25 -0400
Received: from conuserg-08.nifty.com ([10.126.8.71])by condef-07.nifty.com
 with ESMTP id 12VF7UIT002859
        for <kvm@vger.kernel.org>; Thu, 1 Apr 2021 00:07:51 +0900
Received: from localhost.localdomain (133-32-232-101.west.xps.vectant.ne.jp
 [133.32.232.101]) (authenticated)
        by conuserg-08.nifty.com with ESMTP id 12VF77DG021201;
        Thu, 1 Apr 2021 00:07:08 +0900
DKIM-Filter: OpenDKIM Filter v2.10.3 conuserg-08.nifty.com 12VF77DG021201
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=nifty.com;
        s=dec2015msa; t=1617203229;
        bh=X7MGBKWtJ8QFT7mnCJZpDCMgrc+q4Yb5sVuxj/Bmw2c=;
        h=From:To:Cc:Subject:Date:In-Reply-To:References:From;
        b=KPOEIc/aJkyRaiYpnLM+6fyo0GyCuAVQoeVq8nvNcmWZGHY7ButJcwqnkxH11G31j
         RlqJr5Rt00XNwaK3GVj1FScvevsoyLrD4fHfr5RK0bkBrg8+mdw6ydwd9NKK5krqNH
         7JaIqSlmDwZHI7JTxUSt9TSr5zCQpqTc4ZZozyr4WgD62h5E4bkuxzL54tAoTfPp10
         bcZkFOxUFJJGO/EopCNKGkEK5Pp2RhqCeqDu3sTbMD81OOF93UuObL46us9LmtiRa3
         jFXRY3bOVeCEyJc4yNj39mxnDH++yS7eoKhvfT7UXT5aYUNtIz/VcQvMz83ScfG50O
         48xxyBwBKe2Og==
X-Nifty-SrcIP: [133.32.232.101]
From: Masahiro Yamada <masahiroy@kernel.org>
To: linux-mips@linux-mips.org
Cc: Masahiro Yamada <masahiroy@kernel.org>,
        Aleksandar Markovic <aleksandar.qemu.devel@gmail.com>,
        Huacai Chen <chenhuacai@kernel.org>,
        Thomas Bogendoerfer <tsbogend@alpha.franken.de>,
        kvm@vger.kernel.org, linux-kernel@vger.kernel.org,
        linux-mips@vger.kernel.org
Subject: [PATCH 2/2] mips: clean up kvm Makefile
Date: Thu,  1 Apr 2021 00:06:57 +0900
Message-Id: <20210331150658.38919-2-masahiroy@kernel.org>
X-Mailer: git-send-email 2.27.0
In-Reply-To: <20210331150658.38919-1-masahiroy@kernel.org>
References: <20210331150658.38919-1-masahiroy@kernel.org>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

You can use kvm-y instead of kvm-objs to create the composite module.
kvm-$(CONFIG_...) looks cleaner.

Signed-off-by: Masahiro Yamada <masahiroy@kernel.org>
---

 arch/mips/kvm/Makefile | 21 +++++++++------------
 1 file changed, 9 insertions(+), 12 deletions(-)

diff --git a/arch/mips/kvm/Makefile b/arch/mips/kvm/Makefile
index 7d42d624a7b9..6bc5da6faf0e 100644
--- a/arch/mips/kvm/Makefile
+++ b/arch/mips/kvm/Makefile
@@ -2,26 +2,23 @@
 # Makefile for KVM support for MIPS
 #
 
-common-objs-y = $(addprefix ../../../virt/kvm/, kvm_main.o coalesced_mmio.o eventfd.o)
-
 ccflags-y += -Ivirt/kvm -Iarch/mips/kvm
 
-common-objs-$(CONFIG_CPU_HAS_MSA) += msa.o
+kvm-y := $(addprefix ../../../virt/kvm/, kvm_main.o coalesced_mmio.o eventfd.o)
+kvm-$(CONFIG_CPU_HAS_MSA) += msa.o
 
-kvm-objs := $(common-objs-y) mips.o emulate.o entry.o \
+kvm-y +=    mips.o emulate.o entry.o \
 	    interrupt.o stats.o commpage.o \
 	    fpu.o
-kvm-objs += hypcall.o
-kvm-objs += mmu.o
-ifdef CONFIG_CPU_LOONGSON64
-kvm-objs += loongson_ipi.o
-endif
+kvm-y += hypcall.o
+kvm-y += mmu.o
+kvm-$(CONFIG_CPU_LOONGSON64) += loongson_ipi.o
 
 ifdef CONFIG_KVM_MIPS_VZ
-kvm-objs		+= vz.o
+kvm-y		+= vz.o
 else
-kvm-objs		+= dyntrans.o
-kvm-objs		+= trap_emul.o
+kvm-y		+= dyntrans.o
+kvm-y		+= trap_emul.o
 endif
 obj-$(CONFIG_KVM)	+= kvm.o
 obj-y			+= callback.o tlb.o
